examples/spec/integration/call_failing_activity_workflow_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'workflows/call_failing_activity_workflow'
+
+describe CallFailingActivityWorkflow, :integration do
+
+  class TestDeserializer
+    include Temporal::Concerns::Payloads
+  end
+
+  it 'correctly re-raises an activity-thrown exception in the workflow' do
```

Comment:
this is not a very clean spec because it does not isolate activity -> workflow error serialisation. Can we add a unit test for this specific part?

Reply:
(a) There are no existing unit tests of lib/temporal/workflow/state_manager.rb so it's not easy to test in isolation, and 
(b) I need to make sure that the serialization matches the deserialization.  I want a test that encompasses both phases.

Reply:
(a) I'll bootstrap specs for the state manager, so you can piggy back
(b) The main change here is exposing the original error to a workflow. The spec you wrote adds a complexity of workflow result serialization/deserialization using a very different format

Reply:
@antstorm There's a circular dependency.  If you accept this, the test and sample are fixed in https://github.com/coinbase/temporal-ruby/pull/78 .
That's rebased on top of this PR which fixes this test and makes it very clean.
It also shares the same deserialization code at the Activity call as well as polling for the workflow result.
Both changes:
https://github.com/coinbase/temporal-ruby/pull/78/files/b22190eec6d1e2b34d4c2a762a4f946f72ed3b6b..ccdf2457dfee684e237902653a50ff6f9d1e724c